### PR TITLE
Add troubleshoot `ExpansionTile` with PageStorageKey when it has descendant scrollables

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/97583

details: https://github.com/flutter/flutter/issues/97583#issuecomment-1092858939

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: examples/api/lib/material/expansion_tile/expansion_tile.1.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// found in the LICENSE file.
+
+// Flutter code sample for ExpansionTile
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const ExpansionTileApp());
+
+class ExpansionTileApp extends StatelessWidget {
+  const ExpansionTileApp({Key? key}) : super(key: key);
```
Comment (goderbauer) [inline examples/api/lib/material/expansion_tile/expansion_tile.1.dart:12]: nit: you'll have to change these to super parameters. That lint is now enforced for doc samples as well.

File: examples/api/lib/material/expansion_tile/expansion_tile.1.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    return SingleChildScrollView(
+      child: ExpansionTile(
+        key: const PageStorageKey<String>('ExpansionTileKey'),
+        title: const Text('ExpansionTile'),
+        subtitle: const Text('Trailing expansion arrow icon'),
+        children: <Widget>[
+          ListView.builder(
+            key: const PageStorageKey<String>('ListViewKey'),
+            shrinkWrap: true,
```
Comment (goderbauer) [inline examples/api/lib/material/expansion_tile/expansion_tile.1.dart:42]: We are discouraging people from using shrinkWrap, see also https://github.com/flutter/flutter/pull/104008 for context. Why do you need it here? Can this just be a column with mainAxisSize set to min?

File: packages/flutter/lib/src/material/expansion_tile.dart [hunk 3]
```diff
@@ -38,6 +38,29 @@ const Duration _kExpand = Duration(milliseconds: 200);
 /// ** See code in examples/api/lib/material/expansion_tile/expansion_tile.0.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// ### Why do I need to provide unique PageStorageKey(s) to scrollable(s) inside [ExpansionTile]
+/// when providing a PageStorageKey [ExpansionTile]?
+///
+/// When providing a unique PageStorageKey [ExpansionTile] to save and restore its expanded state,
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/expansion_tile.dart:46]: This sentence is super long and hard to parse. Can you split this up into multiple sentences to make it easier to read?

File: packages/flutter/lib/src/material/expansion_tile.dart [hunk 4]
```diff
@@ -38,6 +38,29 @@ const Duration _kExpand = Duration(milliseconds: 200);
 /// ** See code in examples/api/lib/material/expansion_tile/expansion_tile.0.dart **
 /// {@end-tool}
 ///
+/// ## Troubleshooting
+///
+/// ### Why do I need to provide unique PageStorageKey(s) to scrollable(s) inside [ExpansionTile]
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/expansion_tile.dart:43]: This title basically describes a solution. When trouble-shooting you usually start with a problem, though. Maybe state what the problem is in the headline?

File: packages/flutter/lib/src/material/expansion_tile.dart [hunk 5]
```diff
@@ -38,6 +38,29 @@ const Duration _kExpand = Duration(milliseconds: 200);
...
+///
+/// The key provided to the scrollable must be unique for each scrollable.
+///
+/// If PageStorageKey is only provided to [ExpansionTile], it will be used
+/// to restore position of the desendents scrollables of the [ExpansionTile]. While
+/// the [ExpansionTile] uses `bool` value to save and restore the expanded state,
+/// the [Scrollable]s inside [ExpansionTile] use `double` value to save and restore
+/// the scroll offset.
+///
+/// {@tool dartpad}
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/expansion_tile.dart:59]: TBH, it is a little unclear to me what exactly this doc is supposed to tell or teach me...

**Comment by TahaTesser** [comment]
@goderbauer
Thanks for the feedback, updating this soon.

File: packages/flutter/lib/src/material/expansion_tile.dart (hunk 5 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/expansion_tile.dart:59]: You're right, it is a little unclear.
